db/migrations/20230628182641_update_hottest_feed_rename_collection_table.js

```diff
@@ -0,0 +1,111 @@
+const view = `article_hottest_view`
+const materialized = `article_hottest_materialized`
+
+const time_window = 3
+const donation_decay_factor = 0.5
+const boost = 1
+const boost_window = 3
+const matty_donation_decay_factor = 0.95
+const circle_boost = 2
+
+exports.up = async (knex) => {
+  await knex.raw(/*sql*/ `
+  drop view if exists ${view} cascade;
+
+  create view ${view} as
+  with original_score as (
+      select max(read_time_efficiency_boost) as max_efficiency from
+      (
+          select
+          a.id,
+          case when extract(epoch from now()-a.created_at) <= ${boost_window}*3600 then ${boost}*(sum(arc.read_time)::decimal/least(extract(epoch from now()-a.created_at)::decimal + 1, ${time_window}*24*3600))^0.5
+               when ac.article_id is not null and extract(epoch from now()-a.created_at) < 24*3600 then ${circle_boost}*(sum(arc.read_time)::decimal/least(extract(epoch from now()-a.created_at)::decimal + 1, ${time_window}*24*3600))^0.5
+               else (sum(arc.read_time)::decimal/least(extract(epoch from now()-a.created_at)::decimal + 1, ${time_window}*24*3600))^0.5 end as read_time_efficiency_boost
+          from article a
+          join public.user u on a.author_id = u.id
+          join article_read_count arc on a.id = arc.article_id
+          left join article_circle ac on ac.article_id = a.id
+          where a.state = 'active'
+          and arc.created_at >= to_timestamp((extract(epoch from now()) - ${time_window}*24*3600))
+          and arc.user_id is not null
+          and a.id not in (select entrance_id from article_connection where article_id = 8079 )
+          group by a.id, ac.article_id, ac.created_at
+      ) t
+  )
+
+
+  select article.id, article.title, article.created_at, 'https://matters.news/@-/-' || article.media_hash as link, coalesce(t.score, 0) as score
```

Comment:
this `matters.news` can be updated? to `matters.town`; although don't see anywhere is using it

Reply:
fixed